### PR TITLE
Add number of accelerated and delayed modules to SerializerStatistics

### LINKED ISSUE #908: Log number of modules delayed
In `src/serializer/modules.js`, a few things happen.
We should count...
- how many modules get "accelerated" (see `acceleratedModuleIds`).
- how many modules get delayed (look for `delaying require`. Note that this might happen when NOT `isTopLevelRequire`, and then the effect might be undone, and in that case we shouldn't include the delaying in the final count).

The counts should be added to the `SerializerStatistics`, which needs to get passed through to the modules logic.

### PR DESCRIPTION
**Pull request by tcirstea**
Resolves #908.

### CODE REVIEW

File: src/serializer/modules.js [hunk 1]
```diff
@@ -207,6 +213,7 @@ export class ModuleTracer extends Tracer {
 
             if (effects === undefined) {
               console.log(`delaying require(${moduleIdValue})`);
+              if (this.logStatistics) this.statistics.delayedModules++;
```
Comment (NTillmann) [inline src/serializer/modules.js]: In this case, when we are delaying, we are throwing away any partial effects computed by `evaluateForEffects` above, and in that case, we should not keep around statistics about abandoned nested delayed modules.

Meaning: Please save `this.statistics.delayedModules` before the call to `evaluateForEffects`, and if we get into this `effects === undefined` case, restore the value before  incrementing it.

**Comment by tcirstea** [comment]
How can I effectively test this? The file I'm prepacking gives me `{..."valuesInlined":0,"delayedValues":0,"acceleratedModules":0,"delayedModules":0}`

**Comment by NTillmann** [comment]
Some tests in test/serializer/optimizations involve delaying and acceleration, e.g. require_accelerate.js, require_delay.js.

**Comment by tcirstea** [comment]
Ran the tests. As is stands, the only two tests that accelerate one modules are `require_spec_accelerate_delay.js` and `require_accelerate.js`.
No modules are delayed, and based on my debugging, the code never goes into the `if (effects === undefined)` code block.

Another question, do we want the statistics to be collected regardless of the `logStatistics` variable? If it is set to false, the acc/del statistics will always be 0 right now.

**Comment by NTillmann** [comment]
Always collect statistics. Only print to screen when `logStatistics` is given.

**Pull request closed without merging**